### DISCUSSION TITLE
recommendation of vs code over visual studio

### DISCUSSION THREAD

**Discussion by shaicohen-dot-com** [dodyg/practical-aspnetcore#264] [General]
why do you prefer vs code? 

i've been using visual studio since it was visual interdev ... yes i'm old. I've given vs code a "fair-ish" chance, but it felt like visual studio lite. 

in the interest of slowing my descent into tech-dinosaurism, what am i missing?

thanks
(and thank you for sharing all of this)

_* i ask because i want to know, not because i think you don't_

**Comment by dodyg** [comment]
I don't prefer vs code - I use visual studio on my day to day development.

However I think it is important that people learning ASP.NET Core get introduced to the command line tools and deal with the projects without anything getting in the way. Visual Studio hides a lot of stuff. I rather people use `dotnet watch run` and run on Kestrel than clicking F5 on Visual Studio without knowing what's happening in the background.

Also vs-code works really well on small amount of code like the samples here.

> in the interest of slowing my descent into tech-dinosaurism, what am i missing?

You are missing nothing in terms of C# development on Vs code. Visual Studio is much more superior for real projects.

**Reply by Kralizek** [reply]
It has to be said that OmniSharp can have hiccups making the experience on vscode a bit frustrating

**Reply by dodyg** [reply]
Yup. Sometime it's worse. I had to downgrade to `1.23.1` otherwise it won't work at all on my machine.

https://github.com/OmniSharp/omnisharp-vscode/issues/4090

**Reply by shaicohen-dot-com** [reply]
understood,  thank you. 

--story time--
a few years back, i was starting a new position. first thing, of course, was getting the computer set up => i asked for the license key to visual studio => never got one and my trial ran out => again i asked for the license key to visual studio => i was told to explain my justification for expensive software (visual studio) when free software options (vs code) were available

i remember being being so utterly stunned by that question; by the underlying logic (or lack thereof) required to formulate such a question, i never answered. 

i own a personal visual studio subscription now :)

**Reply by dodyg** [reply]
Damn that's terrible. Yeah I wish the C# experience on vs-code is a lot more solid than it is now. vs-code is the IDE of choice for a lot of people starting programming because it is 'lightweight', Javascript/Typescript friendly and less complicated than a normal IDE. It's not a good look for people trying on .NET ecosystem to have their first impression from vs-code.
